analytics/helpers/clone-repos.js

```diff
@@ -0,0 +1,28 @@
+const fs = require('fs');
+const util = require('util');
+const exec = util.promisify(require('child_process').exec);
+
+async function cloneRepos(repos, reposDir) {
+  if (!fs.existsSync(reposDir)) {
+    fs.mkdirSync(reposDir);
+  }
+
+  const clonePromises = repos.map(({ name, url }) =>
+    exec(`gh repo clone ${url} ${reposDir}/${name}`)
+      .then(() => {
+        console.log(`Repo cloned, name: ${name}`);
+        return name;
+      })
+      .catch(() => {
+        console.log(`Repo clone failed, name: ${name}`);
+      }),
+  );
+
+  try {
+    return Promise.all(clonePromises);
```

Comment:
I wonder if we should throttle/debounce these promises so that they are not all send at the same time? It might create too many concurrent network requests which could lead to problems? There's couple ways throttling could be done:

https://www.npmjs.com/package/promise-throttle
https://lodash.com/docs/4.17.15#debounce
https://stackoverflow.com/a/67271450/4662347

Reply:
Hopefully we have some library already installed which can do this. I would not want to add a separate package json for this yet 😁

Reply:
With lodash you can import the specific library for debounce, it should not be very big I think. Or then we could use this solution which does not use external library: https://stackoverflow.com/a/67271450/4662347

Reply:
- [x] add throttling to repo cloning